source/tests/system/waveform_helpers.h

```diff
@@ -0,0 +1,109 @@
+#ifndef NIDEVICE_GRPC_TESTS_WAVEFORM_HELPERS
+#define NIDEVICE_GRPC_TESTS_WAVEFORM_HELPERS
+
+#include <fstream>
+#include <iostream>
+#include <nlohmann/json.hpp>
+#include <vector>
+
+namespace ni {
+namespace tests {
+namespace system {
+
+template <typename TComplex>
+struct TestWaveform {
+  double t0;
+  double dt;
+  std::vector<TComplex> data;
+};
+
+template <typename TFloat>
+struct TestIQData {
+  double t0;
+  double dt;
+  std::vector<TFloat> I;
+  std::vector<TFloat> Q;
+};
+
+template <typename TFloat, typename TComplex>
+TComplex complex(TFloat real, TFloat imaginary)
+{
+  auto c = TComplex{};
+  c.set_real(real);
+  c.set_imaginary(imaginary);
+  return c;
+}
+
+template <typename TFloat, typename TComplex>
+std::vector<TComplex> complex_array(
+    std::vector<TFloat> reals,
+    std::vector<TFloat> imaginaries)
```

Comment:
You probably got this from me, but these should be `const std::vector&`s.


```cpp
std::vector<TComplex> complex_array(
    const std::vector<TFloat>& reals,
    const std::vector<TFloat>& imaginaries)
```

Reply:
updated